### PR TITLE
Qhana

### PR DESCRIPTION
**Pull request by cloudsftp**
Add containers for QHana.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -14,6 +14,7 @@ The defined feature sets are namely:
 - `patternAtlas`
 - `qprov`
 - `winery`
+- `QHana`
```
Comment (salmma) [inline README.md]: First a --> A, adapt everywhere

File: README.md [hunk 2]
```diff
@@ -67,12 +68,17 @@ For running the feature set `winery`, the setup needs additional configuration:
 | QProv-Collector-IBM         | <http://localhost:5021>               | [Link](https://github.com/UST-QuAntiL/qprov)               | [Link](https://hub.docker.com/r/planqk/qprov-collector)                                       |
 | Latex-Renderer              | <http://localhost:5030>               | [Link](https://github.com/UST-QuAntiL/latex-renderer)      | [Link](https://hub.docker.com/repository/docker/planqk/latex-renderer)                        |
 | Redis                       | <http://localhost:5040>               | [Link](https://github.com/redis/redis)                     | [Link](https://hub.docker.com/_/redis)                                                        |
+| Redis (QHana)               | <http://localhost:6379>               | [Link](https://github.com/redis/redis)                     | [Link](https://hub.docker.com/_/redis)                                                        |
```
Comment (salmma) [inline README.md]: Isn't it possible to use the same Redis?

File: README.md [hunk 3]
```diff
@@ -67,12 +68,17 @@ For running the feature set `winery`, the setup needs additional configuration:
...
 | Config-Server               | <http://localhost:2379>               | [Link](https://github.com/etcd-io/etcd)                    | [Link](https://quay.io/repository/coreos/etcd)                                                |
 | Winery                      | <http://localhost:8080>               | [Link](https://github.com/eclipse/winery)                  | [Link](https://hub.docker.com/r/opentosca/winery)                                             |
-| QHAna Plugin Runner         |<http://localhost:5005>                | [Link](https://github.com/UST-QuAntiL/qhana-plugin-runner) | [Link](https://github.com/UST-QuAntiL/qhana-plugin-runner/pkgs/container/qhana-plugin-runner) |
+| QHAna Plugin Runner (Prio)  |<http://localhost:5005>                | [Link](https://github.com/UST-QuAntiL/qhana-plugin-runner) | [Link](https://github.com/UST-QuAntiL/qhana-plugin-runner/pkgs/container/qhana-plugin-runner) |
+| QHAna Plugin Runner         |<http://localhost:5006>                | [Link](https://github.com/UST-QuAntiL/qhana-plugin-runner) | [Link](https://github.com/UST-QuAntiL/qhana-plugin-runner/pkgs/container/qhana-plugin-runner) |
```
Comment (salmma) [inline README.md]: again, is is required to use several plugin runners?

File: README.md [hunk 4]
```diff
@@ -67,12 +68,17 @@ For running the feature set `winery`, the setup needs additional configuration:
...
 | PlanQK Library              |<http://localhost:2903>                | [Link](https://github.com/UST-QuAntiL/PlanQK-Library)      | [Link](https://hub.docker.com/repository/docker/planqk/planqk-library)                        |
 | Gateway                     |<http://localhost:6473>                | [Link](https://github.com/UST-QuAntiL/Gateway)             | [Link](https://hub.docker.com/repository/docker/planqk/gateway)                               |
+| QHana-UI                    | <http://localhost:4200>               | [Link](https://github.com/UST-QuAntiL/qhana-ui)       | [Link](https://github.com/UST-QuAntiL/qhana-plugin-runner/pkgs/container/qhana-ui)|
+| NISQ-Analyzer-UI            | <http://localhost:4201>               | [Link](https://github.com/UST-QuAntiL/nisq-analyzer-ui)       | [Link](https://github.com/UST-QuAntiL/qhana-ui/pkgs/container/qhana-ui)|
```
Comment (salmma) [inline README.md]: please place the NISQ-Analyzer-UI under the NISQ Analyzer component, and use port 5009

File: README.md [hunk 5]
```diff
@@ -81,15 +87,20 @@ For running the feature set `winery`, the setup needs additional configuration:
 * `1978`
 * `2903`
 * `2379`
+* `4200`
```
Comment (salmma) [inline README.md]: avoid standard ports instead group related components together, eg 5000< QHAna ports < 5009

File: docker-compose.yml [hunk 6]
```diff
@@ -349,11 +349,95 @@ services:
...
+      - "6379:6379"
+    networks:
+      - default
+  muse-db:
+    image: ghcr.io/ust-quantil/mini-muse:main
+    profiles: ["all", "QHana"]
+    networks:
+      - default
+
+  nisq-analyzer-ui:
```
Comment (salmma) [inline docker-compose.yml]: also here place nisq analyzer ui next to the nisq analyzer

**Review by salmma — CHANGES_REQUESTED** [review]
File q can be removed? 
See other comments

File: docker-compose.yml [hunk 7]
```diff
@@ -128,7 +128,7 @@ services:
       - "1978:80"
   nisq-analyzer:
     image: planqk/nisq-analyzer:latest
-    profiles: ["all", "nisqAnalyzer", "nisqAnalyzerCompilerComparison"]
+    profiles: ["all", "nisqAnalyzer", "nisqAnalyzerCompilerComparison", "QHana"]
```
Comment (salmma) [inline docker-compose.yml]: let the profile lower case --> qhana

**Review by salmma — CHANGES_REQUESTED** [review]

**Review by salmma — APPROVED** [review]
LGTM

File: README.md [hunk 8]
```diff
@@ -67,10 +69,14 @@ For running the feature set `winery`, the setup needs additional configuration:
 | QProv-Collector-IBM         | <http://localhost:5021>               | [Link](https://github.com/UST-QuAntiL/qprov)               | [Link](https://hub.docker.com/r/planqk/qprov-collector)                                       |
 | Latex-Renderer              | <http://localhost:5030>               | [Link](https://github.com/UST-QuAntiL/latex-renderer)      | [Link](https://hub.docker.com/repository/docker/planqk/latex-renderer)                        |
 | Redis                       | <http://localhost:5040>               | [Link](https://github.com/redis/redis)                     | [Link](https://hub.docker.com/_/redis)                                                        |
+| Redis (QHana)               | <http://localhost:6379>               | [Link](https://github.com/redis/redis)                     | [Link](https://hub.docker.com/_/redis)                                                        |
```
Comment (buehlefs) [inline README.md]: Remove this redis line

File: README.md [hunk 9]
```diff
@@ -67,10 +69,14 @@ For running the feature set `winery`, the setup needs additional configuration:
...
 | Winery                      | <http://localhost:8080>               | [Link](https://github.com/eclipse/winery)                  | [Link](https://hub.docker.com/r/opentosca/winery)                                             |
-| QHAna Plugin Runner         |<http://localhost:5005>                | [Link](https://github.com/UST-QuAntiL/qhana-plugin-runner) | [Link](https://github.com/UST-QuAntiL/qhana-plugin-runner/pkgs/container/qhana-plugin-runner) |
+| QHana-UI                    | <http://localhost:5003>               | [Link](https://github.com/UST-QuAntiL/qhana-ui)       | [Link](https://github.com/UST-QuAntiL/qhana-plugin-runner/pkgs/container/qhana-ui)|
+| QHana-Backend               | <http://localhost:5004>               | [Link](https://github.com/UST-QuAntiL/qhana-backend)       | [Link](https://github.com/UST-QuAntiL/qhana-backend/pkgs/container/qhana-backend)|
+| QHAna Plugin Runner (Prio)  |<http://localhost:5005>                | [Link](https://github.com/UST-QuAntiL/qhana-plugin-runner) | [Link](https://github.com/UST-QuAntiL/qhana-plugin-runner/pkgs/container/qhana-plugin-runner) |
```
Comment (buehlefs) [inline README.md]: There is no longer a separate plugin runner/worker. Please adapt readme to reflect changes.

**Review by buehlefs — CHANGES_REQUESTED** [review]

**Review by buehlefs — APPROVED** [review]

**Pull request merged**